kotlin/src/main/java/org/apache/zeppelin/kotlin/KotlinInterpreter.java

```diff
@@ -0,0 +1,189 @@
+    out = new InterpreterOutputStream(logger);
+  }
+
+  @Override
+  public void close() {
+
+  }
+
+  @Override
+  public InterpreterResult interpret(String code,
+                                     InterpreterContext context) throws InterpreterException{
+    // saving job's running thread for cancelling
+    Job<?> runningJob = getRunningJob(context.getParagraphId());
+    if (runningJob != null) {
+      runningJob.info().put("CURRENT_THREAD", Thread.currentThread());
+    }
+
+    return runWithOutput(code, context.out);
+  }
+
+  @Override
+  public void cancel(InterpreterContext context) throws InterpreterException {
+    Job<?> runningJob = getRunningJob(context.getParagraphId());
+    if (runningJob != null) {
+      Map<String, Object> info = runningJob.info();
+      Object object = info.get("CURRENT_THREAD");
+      if (object instanceof Thread) {
+        try {
+          Thread t = (Thread) object;
+          t.interrupt();
+        } catch (Throwable t) {
+          logger.error("Failed to cancel script: " + t, t);
+        }
+      }
+    }
+  }
+
+  @Override
+  public FormType getFormType() throws InterpreterException {
+    return FormType.NATIVE;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) throws InterpreterException {
+    return 0;
+  }
+
+  @Override
+  public List<InterpreterCompletion> completion(String buf, int cursor,
+      InterpreterContext interpreterContext) throws InterpreterException {
+    return completer.completion(buf, cursor, interpreterContext);
+  }
+
+  public List<KotlinVariableInfo> getVariables() {
+    return interpreter.getVariables();
+  }
+
+  public List<KotlinFunctionInfo> getFunctions() {
+    return interpreter.getFunctions();
+  }
+
+  private Job<?> getRunningJob(String paragraphId) {
+    Job foundJob = null;
+    Collection<Job> jobsRunning = getScheduler().getAllJobs();
+    for (Job job : jobsRunning) {
+      if (job.getId().equals(paragraphId)) {
+        foundJob = job;
+      }
+    }
+    return foundJob;
+  }
+
+  /**
+   * Kotlin interpreter uses System.out for printing, so it is redirected to InterpreterOutput.
+   * Note that Scala's Console class needs separate output redirection
+   */
+  private InterpreterResult runWithOutput(String code, InterpreterOutput out) {
+    this.out.setInterpreterOutput(out);
+
+    PrintStream oldOut = System.out;
+    PrintStream newOut = (out != null) ? new PrintStream(out) : null;
+    try {
+      System.setOut(newOut);
+      return interpreter.eval(code);
+    } finally {
+      System.setOut(oldOut);
+    }
+  }
+
+  private List<String> getImportClasspath(String localRepo) {
+    List<String> classpath = new ArrayList<>();
+    if (localRepo.equals("")) {
+      return classpath;
+    }
+
+    File repo = new File(localRepo);
+    File[] files = repo.listFiles();
+    if (files == null) {
+      return classpath;
+    }
+    for (File file : files) {
+      if (!file.isDirectory()) {
```

Comment:
`file.isFile()` seems to be more intuitive

Reply:
fixed